src/plonkish/compiler/mod.rs

```diff
@@ -54,11 +54,20 @@ pub fn compile<
     }
 }
 
-pub struct PlonkishCompilationResult<F, TG> {
+pub struct PlonkishCompilationResult<F: Clone, TG: TraceGenerator<F> + Clone> {
```

Comment:
@alxkzmn I don't think you need to make the generic types Clone here, just in the impl clause that needs that